src/v/cloud_storage/segment_chunk_api.cc

```diff
@@ -226,6 +179,67 @@ ss::future<> segment_chunks::trim_chunk_files() {
     _eviction_timer.rearm(_eviction_jitter());
 }
 
+ss::future<> segment_chunks::run_hydrate_bg() {
+    auto g = _gate.hold();
+    vassert(_started, "chunk API is not started");
+
+    while (!_gate.is_closed()) {
+        try {
+            co_await _bg_cvar.wait(
+              [this] { return downloads_in_progress() || _gate.is_closed(); });
+            // TODO prioritize "real" requests over prefetches
+            co_await ss::max_concurrent_for_each(
+              std::views::keys(_chunks),
+              _segment.concurrency(),
+              [this](auto start) { return do_hydrate_chunk(start); });
+        } catch (...) {
+            const auto& ex = std::current_exception();
+            for (auto& chunk : std::views::values(_chunks)) {
+                if (!chunk.waiters.empty()) {
+                    set_waiter_errors(chunk.waiters, ex);
+                }
+            }
+            if (ssx::is_shutdown_exception(ex)) {
+                vlog(_ctxlog.debug, "Chunk API hydration loop shut down");
+                break;
+            } else {
+                vlog(
+                  _ctxlog.error, "Chunk API error in hydration loop: {}", ex);
+            }
+        }
+    }
+}
+
+ss::future<>
+segment_chunks::do_hydrate_chunk(chunk_start_offset_t start_offset) {
+    auto g = _gate.hold();
+    vassert(
+      _chunks.contains(start_offset),
+      "No chunk starting at offset {}, cannot hydrate",
+      start_offset);
+
+    auto& chunk = _chunks[start_offset];
+    auto& waiters = chunk.waiters;
+    if (waiters.empty()) {
```

Comment:
Should this skip already hydrated chunks?

Reply:
When we serve requests in `segment_chunks::hydrate_chunk` which are eventually routed to this code path, we do an early return of the chunk file handle for hydrated chunks, and we only progress to enqueue the request where the chunk is not yet hydrated

https://github.com/redpanda-data/redpanda/blob/a9d0662b0fcb6c971f74a3749cba752b9d3884a5/src/v/cloud_storage/segment_chunk_api.cc#L131-L137

There is a vassert before setting chunk handle that the chunk should not already have a file handle associated to it, maybe there can be an added vassert that the chunk state is `download_in_progress`